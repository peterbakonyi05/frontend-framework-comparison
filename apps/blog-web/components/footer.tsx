import React from 'react';
import { Box } from '@chakra-ui/react';

export const Footer = () => {
  return (
    <Box
      as="footer"
      p={4}
      backgroundColor="gray.900"
      color="white"
      textAlign="center"
    >
      &copy; {new Date().getFullYear()} - Simple blog about frontend frameworks
    </Box>
  );
};
