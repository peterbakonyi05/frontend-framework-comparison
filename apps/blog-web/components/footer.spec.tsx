import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from './footer';

describe('Footer', () => {
  beforeEach(() => {
    jest.useFakeTimers().setSystemTime(new Date('2023-06-15'));
  });

  afterEach(() => {
    jest.useRealTimers(); // Restore real timers after each test
  });

  it('should render correctly', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
