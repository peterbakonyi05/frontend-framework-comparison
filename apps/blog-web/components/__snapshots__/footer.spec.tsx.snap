// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<footer
  class="css-h89chm"
>
  Â© 
  2023
   - Simple blog about frontend frameworks
</footer>
`;
